### PR TITLE
docs: joins

### PR DESCRIPTION
**Pull request by soyeric128**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- full outer join
- left semi/anti join
- right semi/anti join

Fixes #issue

### CODE REVIEW

File: docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md [hunk 1]
```diff
@@ -2,17 +2,52 @@
...
+* [Right Join](#right-join)
+* [Full Outer Join](#full-outer-join)
+* [Left / Right Semi Join](#left--right-semi-join)
+* [Left / Right Anti Join](#left--right-anti-join)
+
+## Example Tables
+
+Unless explicitly specified, the join examples on this page are created based on the following tables:
+
+Table "vip_info": This table stores the VIP client information.
```
Comment (xudong963) [inline docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md:22]: 👍, nice to see the example put in the front.

File: docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md [hunk 2]
```diff
@@ -178,97 +174,241 @@ Output:
...
+103|Vancouver
+NULL|NULL
+```
 
-The following command returns all vip_info of all purchase_records, the vip_info will be NULL if purchase_record does not have the corresponding vip_info.
+## Full Outer Join
 
-```sql    
-select vip_info.Client_ID, vip_info.Region from vip_info right join purchase_records on vip_info.Client_ID = purchase_records.Client_ID;
+The *full outer join* returns all rows from both tables, matching up the rows wherever a match can be made and placing NULL s in the places where no matching row exists.
```
Comment (xudong963) [inline docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md:258]: `NULL s` ?

File: docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md (hunk 1 above)
Comment (soyeric128) [reply docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md:22]: Thank you

File: docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md (hunk 2 above)
Comment (soyeric128) [reply docs/doc/30-reference/30-sql/20-query-syntax/dml-join.md]: sorry, typo

**Review by xudong963 — APPROVED** [review]

**Pull request merged**